test/test_http11.py

```diff
@@ -511,6 +512,13 @@ def read(self, size):
         assert 'File-like bodies must return bytestrings. ' \
                'Got: {}'.format(int) in str(exc_info)
 
+    def test_exception_raised_on_socket_timeout(self):
+        c = HTTP11Connection('httpbin.org', socket_timeout=0.01)
```

Comment:
Generally it's good to set this value comedically small so that running on faster machines doesn't accidentally get this to work.